css/src/components/badge.scss

```diff
@@ -71,10 +76,34 @@ $badge-border-radius: $border-radius-rounded !default;
 		}
 	}
 
-	.icon {
+	&.badge-icon {
+		padding: 0;
+		border: none;
 		font-size: $badge-icon-font-size;
 	}
 
+	&.badge-icon-xs {
```

Comment:
Sizing should be at the top level.

.badge
.badge.badge-xs

Reply:
updated